scripts/unicode.py

```diff
@@ -123,13 +119,25 @@ def load_property(filename: str, pattern: str, action: Callable[[int], None]):
                 action(cp)
 
 
-class EffectiveWidth(enum.IntEnum):
-    """Represents the width of a Unicode character. All East Asian Width classes resolve into
-    either `EffectiveWidth.NARROW`, `EffectiveWidth.WIDE`, or `EffectiveWidth.AMBIGUOUS`.
+def to_sorted_ranges(iter: Iterable[Codepoint]) -> list[tuple[Codepoint, Codepoint]]:
```

Comment:
thought: given the amount of logic that is ending up in Python here I think it may make sense to switch to a rust script for this. unfortunately that will make CI much slower since a rust script would need HTTP deps

Perhaps we can instead put more work into documenting and cleaning up the python. you've been doing pretty well so far!

Reply:
> a rust script would need HTTP deps

Or we could just shell out to `curl`…

Reply:
Not opposed.

Reply:
(let's not do that in this PR, though)

Reply:
Actually we could also just check in the fetched file and have a CI job ensure it's up to date.